filebeat/input/docker/input.go

```diff
@@ -45,17 +47,30 @@ func NewInput(
 ) (input.Input, error) {
 	cfgwarn.Experimental("Docker input is enabled.")
 
+	// Wrap log input with custom docker settings
 	config := defaultConfig
 	if err := cfg.Unpack(&config); err != nil {
 		return nil, errors.Wrap(err, "reading docker input config")
 	}
 
-	// Wrap log input with custom docker settings
-	if len(config.Containers.IDs) == 0 {
+	// Docker input should make sure that no callers should ever pass empty strings as container IDs
+	// Hence we explicitly make sure that we catch such things and print stack traces in the event of
+	// an invocation so that it can be fixed.
+	ids := make([]string, 0)
+	for _, containerID := range config.Containers.IDs {
+		if containerID != "" {
+			ids = append(ids, containerID)
+		} else {
+			logp.Err("Docker container ID can't be empty for Docker input config")
+			logp.Debug("docker", "stack trace for empty docker container ID is %v", string(debug.Stack()))
```

Comment:
fyi: The structured logger has support for logging a stack trace. No obligation to change anything - just fyi.

Normally you would create a logger instance and use it for the life of your struct/object, but this code hasn't been update to do this and it's still using the global logger. But the form would be

```
log := logp.NewLogger("docker") 
log.Error("Docker container ID can't be empty for Docker input config")
log.Debugw("Empty Docker container container ID was received", logp.Stack("stacktrace"))
```

or to use the structured logger in an ad-hoc manner

```
logp.L().Named("docker").Debugw("Empty Docker container container ID was received", logp.Stack("stacktrace"))
```